### PR TITLE
macOS support

### PR DESCRIPTION
**Pull request by mTvare6**
Discord uses ~/Library/Application Support to have config files, and just sed doesn't work for macOS, instead use gsed which is GNU/sed for macOS

### CODE REVIEW

File: discocss [hunk 1]
```diff
@@ -34,8 +34,16 @@ EOF
 
 ln -f -s "$preloadFile" /tmp/discocss-preload.js
 
-sed -i 's|  // App preload script, used to provide a replacement native API now that|try { require\(`/tmp/discocss-preload.js`) } catch \(e\) {console.error\(e\);} |' \
+if [ "$(uname)" = "Darwin" ]; then
+    which gsed 1>/dev/null || echo 'run "brew install coreutils" to have GNU/sed for macOS'
+    gsed -i 's|  // App preload script, used to provide a replacement native API now that|try { require\(`/tmp/discocss-preload.js`) } catch \(e\) {console.error\(e\);} |' \
```
Comment (mlvzk) [inline discocss:39]:
```suggestion
    sed -i '' 's|  // App preload script, used to provide a replacement native API now that|try { require\(`/tmp/discocss-preload.js`) } catch \(e\) {console.error\(e\);} |' \
```

Would this work with MacOS' sed?

Comment (mTvare6) [reply discocss:39]: It doesn't work. macOS sed/anytool has always been a pain to work with, so generally devs are suggested to use gsed/g<tool> available thorugh brew or just compiling yourself

Comment (mlvzk) [reply discocss:39]:
```suggestion
    LC_ALL=C sed -i '' 's|  // App preload script, used to provide a replacement native API now that|try { require\(`/tmp/discocss-preload.js`) } catch \(e\) {console.error\(e\);} |' \
```
This should work, I tried it on MacOS 11.2.1.

Comment (mTvare6) [reply discocss:39]: @mlvzk that worked

**Comment by mTvare6** [comment]
ehh, maybe merge?

**Comment by mlvzk** [comment]
@mTvare6 can you refactor the code a bit? Instead of branching into different sed commands, you could have 2 variables, one for sed options, and one for the path to core.asar, and then set them in the if.

the sed would look like
`LC_ALL=C sed $sed_options "$core_asar"`

The check for gsed is also not needed anymore.

**Comment by mlvzk** [comment]
Thanks

**Pull request merged**